File: android/app/src/main/java/it/curzel/bitscape/engine/GameState.kt
package it.curzel.bitscape.engine

import it.curzel.bitscape.gamecore.DisplayableMessage
import it.curzel.bitscape.gamecore.DisplayableToast
import it.curzel.bitscape.gamecore.MatchResult

data class GameState(
    val toasts: DisplayableToast?,
    val messages: DisplayableMessage?,
    val kunai: Int,
    val isInteractionAvailable: Boolean,
    val matchResult: MatchResult,
    val heroHp: Float,
    val isSwordEquipped: Boolean,
    val hasRequestedFastTravel: Boolean
) {
    fun isGameOver(): Boolean {
        return matchResult.gameOver
    }

    fun shouldPauseGame(): Boolean {
        return isGameOver() || messages != null
    }
}

File: ios/SneakBit/Engine/GameState.swift
import Foundation

struct GameState {
    let toasts: CToast
    let messages: CDisplayableMessage
    let kunai: Int32
    let isInteractionAvailable: Bool
    let matchResult: CMatchResult
    let heroHp: Float32
    let isSwordEquipped: Bool
    let hasRequestedFastTravel: Bool
    let hasRequestedPvpArena: Bool
    let currentPlayerIndex: UInt
    
    func isGameOver() -> Bool {
        matchResult.game_over
    }
    
    func shouldPauseGame() -> Bool {
        isGameOver() || messages.is_valid || hasRequestedFastTravel || hasRequestedPvpArena
    }
}
